### PR TITLE
Implement 'couch_epi:is_defined_by_app/4'

### PR DESCRIPTION
**Pull request by iilyak**

### CODE REVIEW

File: src/couch_epi.erl [hunk 1]
```diff
@@ -164,3 +165,11 @@ all(Handle, ServiceId, Function, Args, Opts) ->
 
 is_configured(Handle, Function, Arity) ->
     [] /= couch_epi_functions_gen:modules(Handle, Function, Arity).
+
+-spec is_defined_by_app(
+    Handle :: atom(), SourceApp :: atom(),
+    Function :: atom(), Arity :: pos_integer()) ->
+        list().
```
Comment (kxepal) [inline src/couch_epi.erl]: Returned type is not a list, but boolean.

File: src/couch_epi_functions_gen.erl [hunk 2]
```diff
@@ -61,6 +63,23 @@ apply(Handle, _ServiceId, Function, Args, Opts) ->
...
+    try
+        ModulesBySource = [
+            M || {M, _} <- get_current_definitions(Handle, SourceApp)
+        ],
+        ModulesByFunc = Handle:providers(Function, Arity),
+        lists:filter(fun(I) ->
+            lists:member(I, ModulesBySource)
+        end, ModulesByFunc)
+    catch
+        error:undef -> false
```
Comment (kxepal) [inline src/couch_epi_functions_gen.erl]: What's the point to have multiple returned types? If you defined `list()` in spec, I believe, here empty list should be returned.

Comment (kxepal) [inline src/couch_epi_functions_gen.erl]: Also, I think you have here a bug: if undef error raised, in `is_defined_by_app/4` you'll compare `[] /= false` which give you `true`, but don't think that's right answer,

File: src/couch_epi.erl [hunk 3]
```diff
@@ -21,12 +21,13 @@
     by_key/1, by_key/2, by_source/1, by_source/2,
     keys/1, subscribers/1]).
 
+-export([is_configured/3]).
```
Comment (kxepal) [inline src/couch_epi.erl:24]: Accident export?

File: src/couch_epi.erl [hunk 4]
```diff
@@ -21,12 +21,13 @@
...
+-export([is_defined_by_app/4]).
+
 %% apply
 -export([apply/5]).
 -export([any/5, all/5]).
 
--export([is_configured/3]).
```
Comment (iilyak) [inline src/couch_epi.erl:28]: > @kxepal: Accident export?

No. I just moved the export up. So it is in `%% queries and introspection` section as it should be.

Comment (kxepal) [inline src/couch_epi.erl:28]: Ah, ok. I miss the left column in diff (:

**Comment by kxepal** [comment]
@iilyak Could you rebase and sqash commits here?

**Comment by iilyak** [comment]
stale PR

**Pull request closed without merging**